skbio/io/format/blast6.py

```diff
@@ -0,0 +1,287 @@
+1     0   0.045      15.0
+
+An example of using _blast6_to_data_frame with custom column labels.
+
+>>> from io import StringIO
+>>> from skbio import read
+>>> import pandas as pd
+>>> fs = '\\n'.join([
+...     'subject2\\tquery1\\t100\\tN/A\\t0\\tsubject2\\t32\\t100.0',
+...     'subject1\\tquery2\\t70\\tN/A\\t0\\tsubject1\\t19\\t70.0',
+...     'subject2\\tquery1\\t100\\tN/A\\t0\\tsubject2\\t18\\t50.0'
+... ])
+>>> fh = StringIO(fs)
+
+Load BLAST+6 data into a DataFrame and specify your column types.
+
+>>> df = read(fh, format="blast+6", into=pd.DataFrame,
+...           columns=['sacc', 'qaccver', 'qcovs', 'sblastnames', 'gapopen',
+...                    'sallacc', 'score', 'ppos'])
+>>> df # doctest: +NORMALIZE_WHITESPACE
+       sacc qaccver  qcovs  sblastnames  gapopen   sallacc  score  ppos
+0  subject2  query1    100          NaN        0  subject2     32   100
+1  subject1  query2     70          NaN        0  subject1     19    70
+2  subject2  query1    100          NaN        0  subject2     18    50
+
+References
+----------
+.. [1] http://blastedbio.blogspot.com/2014/11/column-headers-in-blast-tabular-\
+and-csv.html
+.. [2] http://www.ncbi.nlm.nih.gov/books/NBK279675/
+"""
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from skbio.io import create_format
+import pandas as pd
+import numpy as np
+
+blast6 = create_format('blast+6')
+
+_possible_columns = {'qseqid': str, 'qgi': int, 'qacc': str, 'qaccver': str,
+                     'qlen': int, 'sseqid': str, 'sallseqid': str, 'sgi': int,
+                     'sallgi': int, 'sacc': str, 'saccver': str,
+                     'sallacc': str, 'slen': int, 'qstart': int, 'qend': int,
+                     'sstart': int, 'send': int, 'qseq': str, 'sseq': str,
+                     'evalue': float, 'bitscore': float, 'score': int,
+                     'length': int, 'pident': float, 'nident': int,
+                     'mismatch': int, 'positive': int, 'gapopen': int,
+                     'gaps': int, 'ppos': float, 'frames': str, 'qframe': int,
+                     'sframe': int, 'btop': int, 'staxids': str,
+                     'sscinames': str, 'scomnames': str, 'sblastnames': str,
+                     'sskingdoms': str, 'stitle': str, 'salltitles': str,
+                     'sstrand': str, 'qcovs': int, 'qcovhsp': int}
+_default_columns = ['qseqid', 'sseqid', 'pident', 'length', 'mismatch',
+                    'gapopen', 'qstart', 'qend', 'sstart', 'send',
+                    'evalue', 'bitscore']
+
+
+@blast6.reader(pd.DataFrame)
```

Comment:
pass `monkey_patch=False` here, otherwise `.read()` is added to `pd.DataFrame`. We want to avoid this because it's a class defined external to scikit-bio.